co-simulation/patch/CarlaUE4.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# CarlaUE4.sh (or CarlaExecutor.sh, linked as CarlaUE4.sh)
+
+# Retrieve the Docker network name of the current host's container
+NETWORK=$(docker inspect --format '{{range $key, $value := .NetworkSettings.Networks}}{{$key}}{{end}}' $(hostname))
+
+# Check if NETWORK is empty
+if [ -z "$NETWORK" ]; then
+  echo "Warning: Could not determine Docker network. Falling back to 'host' network."
+  NETWORK="host"
+fi
+
+# Run the Docker command with the dynamically determined network
+docker run --privileged --gpus all --name carla-server --net="$NETWORK" --env=DISPLAY=$DISPLAY \
+  --env=NVIDIA_VISIBLE_DEVICES=all --env=NVIDIA_DRIVER_CAPABILITIES=all \
+  --env=SDL_VIDEODRIVER=x11 --volume="/tmp/.X11-unix:/tmp/.X11-unix:rw" \
+  --volume="/usr/share/vulkan/icd.d:/usr/share/vulkan/icd.d:ro" \
+  carlasim/carla:0.10.0 bash CarlaUnreal.sh -vulkan -nosound "$@"
```

Comment:
So, if the CARLA image isn't available locally when the user runs it, we'll download it at this step correct?

If there's a way for us to force that to happen earlier, or at least recommend in documentation that users do it before running the simulation that might be slightly better.

Reply:
Understood, I've reverted that back